### PR TITLE
Ninja Rework (In Progress(Gone Wrong(Kill Me Now)))

### PR DESCRIPTION
**Pull request by ghost**
Work in progress. Moving everything to events, compressing as much as possible, removing uneccessary junk, then fixing any remaining bugs. If the event is still a problem after this, at least it's a nicer looking problem. Putting this up now because the previous coders who worked on this went out of their fucking way to make this as much of a pain in the ass as possible, and I'll be needing a combination of tips and moral support.

### CODE REVIEW

**Comment by Kierany9** [comment]
The hero we need, but not the one we deserve

**Comment by ghost** [comment]
Note that I haven't actually tried to figure out what's making ninja such a pain in the ass whenever more than one are spawned. Right now I'm just cleaning everything up.

Oh, yeah. Also need an In Progress tag.

**Comment by Swagile** [comment]
maybe its the fact that its suit is one of the weirdest most snow flakey piece of shit coded in like FOREVER

its a legacy issue at this point tbh

File: code/modules/events/ninja/energy_katana.dm [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			target.emag_act(user)
+
+
+//If we hit the Ninja who owns this Katana, they catch it.
+//Works for if the Ninja throws it or it throws itself or someone tries
+//To throw it at the ninja
+/obj/item/weapon/katana/energy/throw_impact(atom/hit_atom)
+	if(ishuman(hit_atom))
+		var/mob/living/carbon/human/H = hit_atom
+		if(istype(H.wear_suit, /obj/item/clothing/suit/space/space_ninja))
```
Comment (Carbonhell) [inline code/modules/events/ninja/energy_katana.dm]: perhaps make an isninja proc/define?

**Comment by ghost** [comment]
Well, people want to code new shit for it, so it has to be fixed. It may not even be the suit. I feel like it's either the way ninjas are selected, or some of the code the older coders wrote about in the readme that they were trying to phase out.

Either way, I'm running a fine toothed comb through everything, I'll get it eventually. One of the steps in this process will be re-evaluating all of the ninja's abilities as well as the suit, and gradually re-writing it from the ground up.

Right now my priority is to get this relocated and compressed code working on at least the same level as the old code.

File: code/modules/events/ninja/energy_katana.dm (hunk 1 above)
Comment (ghost) [reply code/modules/events/ninja/energy_katana.dm]: Wouldn't that be equally roundabout when it comes to people killing the ninja and wearing his suit?

**Comment by KayeArray** [comment]
I wish you the best of luck in this horrid endeavor

**Comment by ghost** [comment]
Code is functional as of now, and can even be merged if someone chooses to do so. I haven't actually gotten around to seeing if merging all this shit fixed the runtimes or not, but I can guarantee that the code runs either the same as it did before or better.

If the code gets merged as a bugfix or whatever, then I'll open a new PR with the actual overhauls. Otherwise I'll just keep working on it with this PR.

**Comment by TheOneAndOnlyCreeperJoe** [comment]
Go looking for runtimes by having multiple space ninjas do various things at the same time.

The biggest runtime issues are when there's multiple.

**Comment by ghost** [comment]
That's the plan. I'll hunt for a gerbil later on today, but the code is stable so far with one ninja.

**Comment by ghost** [comment]
what in the goddamn

**Pull request closed without merging**